agentstack/main.py

```diff
@@ -167,36 +168,48 @@ def main():
             webbrowser.open("https://docs.agentstack.sh/quickstart")
         elif args.command in ["init", "i"]:
             init_project_builder(args.slug_name, args.template, args.wizard)
-        elif args.command in ["run", "r"]:
-            run_project(command=args.function, debug=args.debug, cli_args=extra_args)
-        elif args.command in ['generate', 'g']:
-            if args.generate_command in ['agent', 'a']:
-                if not args.llm:
-                    configure_default_model()
-                generation.add_agent(args.name, args.role, args.goal, args.backstory, args.llm)
-            elif args.generate_command in ['task', 't']:
-                generation.add_task(args.name, args.description, args.expected_output, args.agent)
-            else:
-                generate_parser.print_help()
         elif args.command in ["tools", "t"]:
             if args.tools_command in ["list", "l"]:
                 list_tools()
             elif args.tools_command in ["add", "a"]:
+                conf.assert_project()
                 agents = [args.agent] if args.agent else None
                 agents = args.agents.split(",") if args.agents else agents
                 add_tool(args.name, agents)
             elif args.tools_command in ["remove", "r"]:
+                conf.assert_project()
                 generation.remove_tool(args.name)
             else:
                 tools_parser.print_help()
-        elif args.command in ['export', 'e']:
-            export_template(args.filename)
         elif args.command in ['login']:
             auth.login()
         elif args.command in ['update', 'u']:
             pass  # Update check already done
+
+        # inside project dir commands only
+        conf.assert_project()
+
+        if args.command in ["run", "r"]:
+            run_project(command=args.function, debug=args.debug, cli_args=extra_args)
+        elif args.command in ['generate', 'g']:
+            if args.generate_command in ['agent', 'a']:
+                if not args.llm:
+                    configure_default_model()
+                generation.add_agent(args.name, args.role, args.goal, args.backstory, args.llm)
+            elif args.generate_command in ['task', 't']:
+                generation.add_task(args.name, args.description, args.expected_output, args.agent)
+            else:
+                generate_parser.print_help()
+        elif args.command in ['export', 'e']:
+            export_template(args.filename)
         else:
             parser.print_help()
+
+    except conf.NoProjectException as e:
```

Comment:
This could probably be handled by the existing `except Exception` case just as well, since `NoProjectException` inherits from the base exception. 

The logging PR adds relevant improvements: https://github.com/AgentOps-AI/AgentStack/blob/6081e041217c93b6843a073dbbbac870a6488d11/agentstack/main.py#L221

Reply:
yeah, fair enough. i was just trying to get around also printing this line
```python
print(term_color("An error occurred while running your AgentStack command:", "red"))
```